### PR TITLE
Make es docker image creation more efficient

### PR DESCRIPTION
**Pull request by breskeby**
We want avoid packing and unpacking elasticsearch archives to safe build time.

### CODE REVIEW

File: distribution/docker/build.gradle [hunk 1]
```diff
@@ -296,28 +304,25 @@ void addTransformDockerContextTask(Architecture architecture, DockerBase base) {
   def transformTask = tasks.register(taskName("transform", architecture, base, "DockerContext"), Sync) {
     TaskProvider<Tar> buildContextTask = tasks.named(taskName("build", architecture, base, "DockerContext"))
     dependsOn(buildContextTask)
-
+    duplicatesStrategy = DuplicatesStrategy.EXCLUDE
```
Comment (mark-vieira) [inline distribution/docker/build.gradle:307]: Does the exploded distro include duplicate entires? Why is this needed?

**Review by mark-vieira — APPROVED** [review]
LGTM. This was surprisingly simple to implement. Thanks, Rene!

File: distribution/docker/build.gradle (hunk 1 above)
Comment (breskeby) [reply distribution/docker/build.gradle:307]: This is needed as we have tweaked config files like `config/elasticsearch.yml` and `config/log4j2.properties` and we want to keep them when copying over the elasticsearch distro alternatively we could exclude it from that specific copy definition but that seemed simpler and should also be faster by not checking excludes with every file.

Comment (mark-vieira) [reply distribution/docker/build.gradle:307]: Got it, that just means that order matters here so perhaps we should add some comments to that effect so we don't accidentally mess this up later.

**Review by pugnascotia — APPROVED** [review]
